src/luacheck/fs.lua

```diff
@@ -159,23 +160,20 @@ function fs.extract_files(dir_path, pattern)
                local err = scan(full_path)
 
                if err then
-                  return err
+                  err_map[full_path] = err
+                  table.insert(res, full_path)
                end
+
             elseif path:match(pattern) and fs.is_file(full_path) then
                table.insert(res, full_path)
             end
          end
       end
    end
 
-   local err = scan(dir_path)
-
-   if err then
-      return nil, err
-   end
-
+   scan(dir_path)
```

Comment:
`scan` may return an error if scanning failed for the initial `dir_path`. Could add same return value handling as for the recursive `scan()` call, but I think it would make more sense to put the error into `err_map` as soon as possible (so after the `utils.unprefix` call) and have `scan()` not return anything.